### PR TITLE
Upgrade to Google Cloud Service 2.12.0

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

**Comment by aloubyansky** [comment]
The LanguageTest seems to be a new failure in the CQ testsuite. FYI @jamesnetherton

File: generated-platform-project/pom.xml [hunk 1]
```diff
@@ -73,7 +73,7 @@
           <plugin>
             <groupId>org.sonatype.plugins</groupId>
             <artifactId>nexus-staging-maven-plugin</artifactId>
-            <version>1.6.13</version>
```
Comment (aloubyansky) [inline generated-platform-project/pom.xml:76]: @loicmathieu could you please try re-generating the platform to see if it results in a diff? I just tried upgrading the GCS version, `mvn -Dsync` and I didn't get this change. Thanks.

Comment (loicmathieu) [reply generated-platform-project/pom.xml:76]: Just did it, no change

**Comment by aloubyansky** [comment]
@loicmathieu this is targeting 3.16. Do you also want this in 3.15 (LTS) as well? If so, please open a PR for the 3.15 branch as well. Thanks.

**Comment by gemmellr** [comment]
This appears to have broken the main build. (I also also see generated changes, as per #1282)

If you run ./mvnw -Dsync it completes but does show a warning:
```
[WARNING] The POM for io.quarkiverse.googlecloudservices:quarkus-google-cloud-services-main-it:pom:2.12.0 is missing, no dependency information available
```

However if you run ./mvnw clean install -DskipTests it actually fails. This is also visible in the CI runs from main, e.g snippet from CI logs:
```
2024-09-19T17:21:30.6229966Z [ERROR] Failed to execute goal on project quarkus-google-cloud-services-main-it: Could not resolve dependencies for project io.quarkus.platform:quarkus-google-cloud-services-main-it:jar:999-SNAPSHOT: The following artifacts could not be resolved: io.quarkiverse.googlecloudservices:quarkus-google-cloud-services-main-it:jar:2.12.0 (absent), io.quarkiverse.googlecloudservices:quarkus-google-cloud-services-main-it:jar:tests:2.12.0 (absent): Could not find artifact io.quarkiverse.googlecloudservices:quarkus-google-cloud-services-main-it:jar:2.12.0 in google-maven-central (https://maven-central.storage-download.googleapis.com/maven2/) -> [Help 2]
```

**Comment by aloubyansky** [comment]
@loicmathieu Robbie's right https://repo.maven.apache.org/maven2/io/quarkiverse/googlecloudservices/quarkus-google-cloud-services-main-it/
https://github.com/quarkiverse/quarkus-google-cloud-services/commit/e3fbd511cc86b0f8c59d478aade7cc31b8263510 was not a good idea. Could you please add the test jars back?

**Comment by loicmathieu** [comment]
@aloubyansky will be done by https://github.com/quarkiverse/quarkus-google-cloud-services/pull/675

Should I release a new version and upgrade the platform?

**Comment by loicmathieu** [comment]
> @loicmathieu this is targeting 3.16. Do you also want this in 3.15 (LTS) as well? If so, please open a PR for the 3.15 branch as well. Thanks.

I thought main was 3.15 as 3.15 was released a few days ago, yes it must target also 3.15, I'll open a PR on the 3.15 branch.

**Comment by aloubyansky** [comment]
> @aloubyansky will be done by [quarkiverse/quarkus-google-cloud-services#675](https://github.com/quarkiverse/quarkus-google-cloud-services/pull/675)
> 
> Should I release a new version and upgrade the platform?

That would be great, so we can run proper tests for GCS in the platform.

**Pull request merged**